code/src/test/kotlin/com/expediagroup/sdk/core/authentication/bearer/BearerAuthenticationManagerTest.kt

```diff
@@ -0,0 +1,552 @@
+package com.expediagroup.sdk.core.authentication.bearer
+
+import com.expediagroup.sdk.core.authentication.common.Credentials
+import com.expediagroup.sdk.core.client.RequestExecutor
+import com.expediagroup.sdk.core.http.CommonMediaTypes
+import com.expediagroup.sdk.core.http.Method
+import com.expediagroup.sdk.core.http.Protocol
+import com.expediagroup.sdk.core.http.Request
+import com.expediagroup.sdk.core.http.Response
+import com.expediagroup.sdk.core.http.ResponseBody
+import com.expediagroup.sdk.core.http.Status
+import com.expediagroup.sdk.core.model.exception.client.ExpediaGroupResponseParsingException
+import com.expediagroup.sdk.core.model.exception.service.ExpediaGroupAuthException
+import com.expediagroup.sdk.core.model.exception.service.ExpediaGroupNetworkException
+import io.mockk.clearMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class BearerAuthenticationManagerTest {
+
+    private lateinit var requestExecutor: RequestExecutor
+    private lateinit var credentials: Credentials
+    private lateinit var authenticationManager: BearerAuthenticationManager
+    private val authUrl = "http://auth.example.com/token"
+
+    @BeforeAll
+    fun setup() {
+        requestExecutor = mockk()
+        credentials = Credentials("client_key", "client_secret")
+        authenticationManager = BearerAuthenticationManager(authUrl, requestExecutor, credentials)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        clearMocks(requestExecutor)
+        authenticationManager.clearAuthentication()
+    }
+
+
+    /**
+     * Test successful authentication flow.
+     */
+    @Test
+    fun `authenticate should store token on successful response`() {
```

Comment:
```suggestion
    fun `should authenticate and store access token on successful response`() {
```
It's better to describe the expected behavior in abstraction without tying the test to a specific method name.

Reply:
done.